app/models/derived_variables/sales_log_variables.rb

```diff
@@ -16,5 +16,77 @@ def set_derived_fields!
       self.mscharge = 0
     end
     self.pcode1, self.pcode2 = postcode_full.split(" ") if postcode_full.present?
+    self.totchild = total_child
+    self.totadult = total_adult + total_elder
+    self.hhmemb = totchild + totadult
+    self.hhtype = household_type
+  end
+
+private
+
+  def total_elder
+    ages = [age1, age2, age3, age4, age5, age6]
+    ages.count { |age| age.present? && age >= 60 }
+  end
+
+  def total_child
+    (2..6).count do |i|
+      age = public_send("age#{i}")
+      relat = public_send("relat#{i}")
+      age.present? && ((age < 20 && %w[C].include?(relat)))
+    end
+  end
+
+  def total_adult
+    total = age1.present? && age1.between?(16, 59) ? 1 : 0
+    total + (2..6).count do |i|
+      age = public_send("age#{i}")
+      relat = public_send("relat#{i}")
+      age.present? && ((age.between?(20, 59) || age.between?(18, 19) && relat != "C"))
```

Comment:
Are the double brackets necessary? Or was the second open bracket meant to be before `age.between?(18, 19)` rather than before `age.between?(20, 59)` (presumably in order to help make clearer the order of operation)?